R/gpdd.R

```diff
@@ -1,22 +1,42 @@
- 
 
-#' Download Tables from the GPDD Data
-#' @param table name of the table to be downloaded.
-#' @return the requested data.frame (as a tibble)
-#' @export
-#' @examples 
-#' gpdd("location")
-gpdd <- function(table = c("data", "main", "timeperiod", "location")){
-  
-  table <- match.arg(table)
-  
-  url <- switch(table,
-                location = "https://knb.ecoinformatics.org/knb/d1/mn/v2/object/df35b.239.1",
-                main = "https://knb.ecoinformatics.org/knb/d1/mn/v2/object/df35b.234.1")
-    
+#' Download Data from GPDD.
+#'
+#' @author Zihao Li, \email{zihao0104@@berkeley.edu}
+#' @param dataset_name A string of the name of the dataset.
+#' @return The requested dataset in tibble format.
+#' @examples
```

Comment:
You need to use `@importFrom readr read_csv`  and you need to run `usethis::use_package("readr")`.  This will add it as a dependency to the package.  We do not ask users to manage dependencies 'manually', a package should know how to install it's own dependencies when documented in this way.

Reply:
This is done.